lib/samples/show_viewshed_from_geoelement_in_scene/show_viewshed_from_geoelement_in_scene.dart

```diff
@@ -0,0 +1,288 @@
+
+      final tempPos = _tankGraphic!.geometry! as ArcGISPoint;
+
+      final currentPos = ArcGISPoint(
+        x: tempPos.x,
+        y: tempPos.y,
+        spatialReference: SpatialReference.wgs84,
+      );
+      final target = ArcGISPoint(
+        x: _waypoint!.x,
+        y: _waypoint!.y,
+        spatialReference: SpatialReference.wgs84,
+      );
+
+      // Use geodetic distance to get distance and azimuth
+      final result = GeometryEngine.distanceGeodetic(
+        point1: currentPos,
+        point2: target,
+        distanceUnit: LinearUnit(unitId: LinearUnitId.meters),
+        azimuthUnit: AngularUnit(unitId: AngularUnitId.degrees),
+        curveType: GeodeticCurveType.geodesic,
+      );
+
+      final distance = result.distance;
+      final azimuth = result.azimuth1;
+
+      // Stop if close enough
+      if (distance <= 5) {
+        _waypoint = null;
+        timer.cancel();
+        return;
+      }
+
+      // Move a small step toward the waypoint
+      const step = 1.0; // meters
+      final movedPoints = GeometryEngine.moveGeodetic(
+        pointCollection: [currentPos],
+        distance: step,
+        azimuth: azimuth,
+        distanceUnit: LinearUnit(unitId: LinearUnitId.meters),
+        azimuthUnit: AngularUnit(unitId: AngularUnitId.degrees),
+        curveType: GeodeticCurveType.geodesic,
+      );
+
+      if (movedPoints.isEmpty) return;
+
+      final newPoint = movedPoints.first;
+      _tankGraphic!.geometry = newPoint;
+
+      // Update heading
+      final currentHeading =
+          (_tankGraphic!.attributes['HEADING'] as num?)?.toDouble() ?? 0.0;
+
+      final headingDiff = shortestAngle(currentHeading, azimuth);
+      final newHeading = currentHeading + headingDiff / 10;
+
+      _tankGraphic!.attributes['HEADING'] = newHeading;
+    });
+  }
+
+  // Calculate shortest angle to rotate
+
+  double shortestAngle(double from, double to) {
+    final difference = (to - from + 540) % 360 - 180;
+    return difference;
+  }
+
+  // Loads the 3D tank model from local sample data and returns it as a Graphic.
+  Future<Graphic> _loadTankGraphic() async {
+    await downloadSampleData(['07d62a792ab6496d9b772a24efea45d0']);
+    final documentsDirPath =
+        (await getApplicationDocumentsDirectory()).absolute.path;
+    final tankModelPath = '$documentsDirPath/bradley_low_3ds/bradle.3ds';
+
+    // Define the tank symbol.
+    final tankSymbol =
+        ModelSceneSymbol.withUri(uri: Uri.parse(tankModelPath), scale: 10)
+          ..heading = 90
+          ..anchorPosition = SceneSymbolAnchorPosition.bottom;
+
+    // Return the graphic that combines geometry and symbol.
+    return Graphic(
+      geometry: ArcGISPoint(x: -4.506390, y: 48.385624),
+      attributes: {'HEADING': 0.0},
+      symbol: tankSymbol,
+    );
+  }
+
+  // Adds the tank graphic to a graphics overlay and sets the initial viewpoint.
+  Future<void> _addTankToScene(Graphic tankGraphic) async {
```

Comment:
There's no `await` in this function, so it can be declared more simply `void _addTankToScene(Graphic tankGraphic) {`. And then it doesn't need to be `await`ed where it is called above.